cpp/include/Ice/MetricsAdminI.h

```diff
@@ -200,10 +196,7 @@ namespace IceInternal
             [[nodiscard]] IceMX::MetricsPtr clone() const
             {
                 TPtr metrics = std::dynamic_pointer_cast<T>(_object->ice_clone());
-                for (typename std::map<std::string, std::pair<MetricsMapIPtr, SubMapMember>>::const_iterator p =
-                         _subMaps.begin();
-                     p != _subMaps.end();
-                     ++p)
+                for (auto p = _subMaps.begin(); p != _subMaps.end(); ++p)
```

Comment:
I don't know why -fix didn't convert this code in a loop.

Reply:
Actually, in this case, I prefer to leave it alone as the syntax is hard to decipher.